ReSharper.FSharp/src/FSharp.Psi.Daemon/src/QuickDoc/FSharpQuickDocProvider.fs

```diff
@@ -24,9 +24,16 @@ module FSharpQuickDoc =
 
         // todo: fix getting qualifiers
         let tokenNames = [token.Name]
+        let treeEndOffset =
+            if (isNull token.Parent) then token.GetTreeEndOffset() else
+            if (isNull token.Parent.Parent) then token.GetTreeEndOffset() else
```

Comment:
Going up the tree with `Parent` properties without checking the node types and children roles is very dangerous because it may work in too many places where you may not expect it. Unless you want this check to pass for every identifier that has a parent (and the most of them do indeed have one), you should check what parents are and, where applicable, from which of the children you go up. This is why the generated navigators should be used 95% of the time.

If you're interested in active pattern nodes, the please check that the parent node is an active pattern one. Please check the parsed tree.

Reply:
Please note that `IActivePatternId` is also an `IFSharpIdentifier`, as well as `ITypeParameterId`, so it may worth checking if detecting and passing them instead of the initial inner identifiers could fix it too.